tests/migrations/test_operations.py

```diff
@@ -241,6 +241,274 @@ def test_create_model_with_db_level_fk(self):
         if connection.features.has_on_delete_db_default:
             self.assertColumnExists(f"{app_label}_rider", "pony_default_id")
 
+    def change_to_db_cascade_test(
+        self, app_label, fk_parent_model, fk_child_model, fk_fieldname, project_state
+    ):
+        operation = migrations.AlterField(
+            fk_child_model,
+            fk_fieldname,
+            models.ForeignKey(
+                f"{app_label}.{fk_parent_model}", on_delete=models.DB_CASCADE
+            ),
+        )
+        new_state = project_state.clone()
+        operation.state_forwards(app_label, new_state)
+
+        with connection.schema_editor() as editor:
+            operation.database_forwards(app_label, editor, project_state, new_state)
+
+        Rider = new_state.apps.get_model(app_label, fk_child_model)
+        Pony = new_state.apps.get_model(app_label, fk_parent_model)
+
+        pony = Pony.objects.create()
+        kwargs = {fk_fieldname: pony}
+        Rider.objects.create(**kwargs)
+
+        with self.assertNumQueries(1):
+            pony.delete()
```

Comment:
I see that we're testing that `pony.delete()` only causes one query.

I think we should also test that all the rows that we expect to be deleted *are* deleted.

Reply:
Done over two roes here : https://github.com/django/django/pull/16851/files#diff-7cc40f1cb85a8f19c80b2e901626d78e4729047dbbb0ed52861f5894ff688115R258